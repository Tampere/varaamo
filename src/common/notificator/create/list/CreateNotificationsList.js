import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'react-bootstrap';
import moment from 'moment';

const CreateNotificationsList = (props) => {
  const { notifications, onClick } = props;
  return (
    <div className="notification-list">
      <h4>Notifications list</h4>
      <Table>
        <thead>
          <tr>
            <th>Name</th>
            <th>Created</th>
            <th>Until</th>
            <th>Target</th>
            <th>Urgency</th>
            <th>Active</th>
          </tr>
        </thead>
        <tbody>
          {notifications && notifications.map((notification, index) => (
            <tr key={index} onClick={() => onClick(notification)}>
              <th>{notification.name}</th>
              <th>{moment(notification.created).format('DD.MM.YYYY')}</th>
              <th>{moment(notification.until).format('DD.MM.YYYY')}</th>
              <th>{notification.target.label}</th>
              <th>{notification.urgency.label}</th>
              <th>{notification.active.label}</th>
            </tr>
          ))}
        </tbody>
      </Table>
    </div>
  );
};

CreateNotificationsList.propTypes = {
  notifications: PropTypes.array.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default CreateNotificationsList;
