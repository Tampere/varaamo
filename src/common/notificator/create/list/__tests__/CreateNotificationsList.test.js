import React from 'react';
import toJson from 'enzyme-to-json';
import simple from 'simple-mock';
import { shallow } from 'enzyme';

import CreateNotificationsList from '../CreateNotificationsList';

describe('CreateNotificationList', () => {
  const defaultProps = {
    notifications: [],
    onClick: simple.stub(),
  };
  const getWrapper = props => shallow(<CreateNotificationsList {...defaultProps} {...props} />);

  test('Renders correctly', () => {
    const wrapper = getWrapper();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
