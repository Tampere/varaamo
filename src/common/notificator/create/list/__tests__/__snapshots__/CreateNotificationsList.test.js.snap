// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNotificationList Renders correctly 1`] = `
<div
  className="notification-list"
>
  <h4>
    Notifications list
  </h4>
  <Table
    bordered={false}
    bsClass="table"
    condensed={false}
    hover={false}
    responsive={false}
    striped={false}
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Created
        </th>
        <th>
          Until
        </th>
        <th>
          Target
        </th>
        <th>
          Urgency
        </th>
        <th>
          Active
        </th>
      </tr>
    </thead>
    <tbody />
  </Table>
</div>
`;
